pkg/cloudfoundry/ManifestUtils.go

```diff
@@ -0,0 +1,226 @@
+package cloudfoundry
+
+import (
+	"fmt"
+	"github.com/ghodss/yaml"
+	"github.com/pkg/errors"
+	"io/ioutil"
+	"reflect"
+
+	"github.com/SAP/jenkins-library/pkg/log"
+)
+
+const propApplications = "applications"
+const propBuildpacks = "buildpacks"
+const propBuildpack = "buildpack"
+
+// Manifest ...
+type Manifest struct {
+	self     map[string]interface{}
+	modified bool
+	name     string
+}
+
+var m Manifest
+
+var _readFile = ioutil.ReadFile
+
+// ReadManifest ...
+func ReadManifest(name string) (Manifest, error) {
+
+	log.Entry().Infof("Reading manifest file  '%s'", name)
+
+	m := Manifest{self: make(map[string]interface{}), name: name, modified: false}
+
+	content, err := _readFile(name)
+	if err != nil {
+		return m, errors.Wrapf(err, "cannot read file '%v'", m.name)
+	}
+
+	err = yaml.Unmarshal(content, &m.self)
+
+	if err != nil {
+		return m, errors.Wrapf(err, "Cannot parse yaml file '%s': %s", m.name, string(content))
+	}
+
+	log.Entry().Infof("Manifest file '%s' has been parsed", m.name)
+
+	return m, nil
+}
+
+// WriteManifest ...
+func (m *Manifest) WriteManifest() error {
+
+	d, err := yaml.Marshal(&m.self)
+	if err != nil {
+		return err
+	}
+
+	log.Entry().Debugf("Writing manifest file '%s'", m.name)
+	err = ioutil.WriteFile(m.name, d, 0644)
+
+	if err == nil {
+		m.modified = false
+	}
+
+	log.Entry().Debugf("Manifest file '%s' has been written", m.name)
+	return err
+}
+
+// GetName ...
+func (m Manifest) GetName() string {
+	return m.name
+}
+
+// GetApplications ...
+func (m Manifest) GetApplications() ([]interface{}, error) {
+	return toSlice(m.self)
+}
+
+// ApplicationHasProperty Checks if the application denoted by 'index' has the property 'name'
+func (m Manifest) ApplicationHasProperty(index int, name string) (bool, error) {
+
+	sliced, err := toSlice(m.self[propApplications])
+
+	if err != nil {
+		return false, err
+	}
+
+	if index >= len(sliced) {
+		return false, fmt.Errorf("Index (%d) out of bound. Number of apps: %d", index, len(sliced))
+	}
+
+	_m, err := toMap(sliced[index])
+
+	if err != nil {
+		return false, err
+	}
+
+	_, ok := _m[name]
+
+	return ok, nil
+}
+
+// GetApplicationProperty ...
+func (m Manifest) GetApplicationProperty(index int, name string) (interface{}, error) {
+
+	sliced, err := toSlice(m.self[propApplications])
+
+	if err != nil {
+		return "", err
+	}
+
+	if index >= len(sliced) {
+		return "", fmt.Errorf("Index (%d) out of bound. Number of apps: %d", index, len(sliced))
+	}
+
+	app, err := toMap(sliced[index])
+
+	if err != nil {
+		return "", err
+	}
+
+	if app[name] != nil {
+		return app[name], nil
+	}
+
+	return "", fmt.Errorf("No such property: '%s' available in application at position %d", name, index)
```

Comment:
Why return `""` and not `nil`?

Reply:
> Why return `""` and not `nil`?

The compiler does not allow that: `cannot use nil as type string in return argument`

Reply:
Have you checked? In this method, you return `interface{}` not `string`. If I test it, it works.

Reply:
@stippi2 : you are right, here we can return `nil` since the return type is `interface{}`. I tested it somewhere (... and took the compiler message above from there), but apparently I was puzzled with the location where I tested. --> We return `nil` now.